pulp_rpm/app/shared_utils.py

```diff
@@ -22,8 +24,15 @@ def _prepare_package(artifact, filename):
     # Copy file to a temp directory under the user provided filename
     with tempfile.TemporaryDirectory() as td:
         temp_path = os.path.join(td, filename)
-        shutil.copy2(artifact.file.path, temp_path)
-        cr_pkginfo = createrepo_c.package_from_rpm(temp_path)
+        try:
+            shutil.copy2(artifact.file.path, temp_path)
+            cr_pkginfo = createrepo_c.package_from_rpm(temp_path)
+        except NotImplementedError:
+            artifact_s3_file = storage.open(artifact.file.name)
```

Comment:
Probably shouldn't use `s3` in the name. We also support azure and I think it will use the same code path. I would just leave a comment saying `# handle files not stored on the filesystem` and call the variable `artifact_file`.